frontend/src/components/auth/Settings.tsx

```diff
@@ -0,0 +1,758 @@
+                    <FormControl
+                      isRequired={index === 0}
+                      isInvalid={
+                        attemptedSubmit && onsiteInfo[index].phone === ""
+                      }
+                    >
+                      <Input
+                        h="37px"
+                        type="tel"
+                        value={onsiteInfo[index].phone}
+                        placeholder={PLACEHOLDER_WEB_EXAMPLE_PHONE_NUMBER}
+                        onChange={(e) => {
+                          onsiteInfo[index].phone = e.target.value;
+                          setOnsiteInfo([...onsiteInfo]);
+                        }}
+                      />
+                    </FormControl>
+                  </Td>
+                  <Td padding="0 0 0 12px">
+                    <FormControl
+                      isRequired={index === 0}
+                      isInvalid={
+                        attemptedSubmit &&
+                        !isValidEmail(onsiteInfo[index].email)
+                      }
+                    >
+                      <Input
+                        h="37px"
+                        type="email"
+                        value={onsiteInfo[index].email}
+                        placeholder={PLACEHOLDER_WEB_EXAMPLE_EMAIL}
+                        onChange={(e) => {
+                          onsiteInfo[index].email = e.target.value;
+                          setOnsiteInfo([...onsiteInfo]);
+                        }}
+                      />
+                    </FormControl>
+                  </Td>
+                  {onsiteInfo.length >= 2 ? (
+                    <Td padding="0 4px">
+                      <DeleteIcon
+                        h="19.5px"
+                        w="100%"
+                        color="gray.gray300"
+                        cursor="pointer"
+                        _hover={{ color: "primary.blue" }}
+                        onClick={() => {
+                          onsiteInfo.splice(index, 1);
+                          setOnsiteInfo([...onsiteInfo]);
+                        }}
+                      />
+                    </Td>
+                  ) : (
+                    <Td padding="0 4px" />
+                  )}
+                </Tr>
+              ))}
+            </Tbody>
+          </Table>
+        </TableContainer>
+        {onsiteInfo.length < 10 && (
+          <Text
+            variant="desktop-button-bold"
+            color="primary.blue"
+            cursor="pointer"
+            w="fit-content"
+            onClick={() => {
+              setOnsiteInfo([
+                ...onsiteInfo,
+                {
+                  name: "",
+                  phone: "",
+                  email: "",
+                },
+              ]);
+            }}
+          >
+            + Add another contact
+          </Text>
+        )}
+      </Flex>
+    );
+  };
+
+  const getMobileOnsiteStaffSection = (): React.ReactElement => {
```

Comment:
This is exactly the same as the form for the sign up? If so, could we move `getWebOnsiteStaffSection` and `getMobileOnsiteStaffSection` to a separate component (maybe `components/common/OnsiteStaffForm.tsx`). This way we can share it between the sign up page and here instead of copying it. Furthermore, a few other pages also need the same form, so it will cut down on future effort as well.

I recommend taking the `onsiteInfo`, and `setOnsiteInfo` as props for the new component. E.g. we'll be able to use it as 
`<OnsiteStaffForm onsiteInfo={onsiteInfo} setOnsiteInfo={setOnsiteInfo}/>` 

Please also swap out the sign up form to use the new component.

If you guys disagree or if this is challenging LMK, but I think extracting it will be helpful for the future.